functional_tests/scylla_operator/test_functional.py

```diff
@@ -564,6 +564,30 @@ def test_deploy_helm_with_default_values(db_cluster: ScyllaPodCluster):
         db_cluster.k8s_cluster.kubectl(f"delete namespace {namespace}")
 
 
+@pytest.mark.restart_is_used
+def test_rolling_config_change_internode_compression(db_cluster, scylla_yaml):
+    """
+    Cover logic of disrupt_rolling_config_change_internode_compression nemesis
+    """
+    def get_internode_compression_new_value_randomly():
+        log.debug("Current compression is %s", original_compression)
+        values = ['dc', 'all', 'none']
+        values_to_toggle = list(filter(lambda value: value != original_compression, values))
+        return random.choice(values_to_toggle)
+
+    internode_compression_option_name = "internode_compression"
+
+    with db_cluster.nodes[0].actual_scylla_yaml() as props:
+        original_compression = dict(props).get(internode_compression_option_name)
```

Comment:
why are you getting `original_compression` ? it's not that you are using it later ?

i.e. we might want to revert it back to what was set, don't we ?

Reply:
now I see it's used inside `get_internode_compression_new_value_randomly`

I wouldn't recommend  using different function context like that, especially when the function isn't written inline.
it's not clear to whom reads it, and when someone might refactor and move this function it would be missing that variable.

anyhow since `get_internode_compression_new_value_randomly`  is only called once, maybe it can be embedded into the test logic.

Reply:
changed